File: 1/2021.03.26/116.Populating Next Right Pointers in Each Node/java/solution.java
class Solution {
    public Node connect(Node root) {
        if (root != null) {
            connectTwo(root.left, root.right);
        }
        return root;
    }

    private void connectTwo(Node n1, Node n2) {
        if (n1 == null || n2 == null) {
            return;
        }
        n1.next = n2;
        connectTwo(n1.left, n1.right);
        connectTwo(n1.right, n2.left);
        connectTwo(n2.left, n2.right);
    }
}

File: 1/2021.03.26/116.Populating Next Right Pointers in Each Node/python/solution.py
class Solution:
    def connect(self, root: 'Node') -> 'Node':
        # level-order traversal
        if not root:
            return
        q = [root]
        while q:
            size = len(q)
            pre = None
            for _ in range(size):
                node = q.pop(0)
                if pre is not None:
                    pre.next = node
                pre = node
                if node.left:
                    q.append(node.left)
                    q.append(node.right)
        return root
